DOCS-11473 added content for retry mechanism (#387)
mulesoft/docs-api-manager: modules/ROOT/pages/policy-mule4-jwt-validation.adoc

`#[authentication.properties.jwt]`

=== Retry Mechanism

> If you selected JSON Web Key Sets (JWKS) as JWT Key Origin, the policy uses a retry mechanism to handle failed JWKS fetches. Every time the cached JWKS expires or is absent and a new request is initiated, a request is sent to the JWKS service. If the request fails, a sequence of asynchronous requests with an exponential backoff delay are sent to the JWKS service until a new JWKS is obtained and cached again. 

Reviewer: "a sequence of...is sent":
```suggestion
If you selected JSON Web Key Sets (JWKS) as JWT Key Origin, the policy uses a retry mechanism to handle failed JWKS fetches. Every time the cached JWKS expires or is absent and a new request is initiated, a request is sent to the JWKS service. If the request fails, a sequence of asynchronous requests with an exponential backoff delay is sent to the JWKS service until a new JWKS is obtained and cached again. 
```

Author: Done! Thanks!

> Until the recaching completes, the expired JWKS is used to process the request, and a `503` error is returned if no request is found. This enables you to use a cached value in upcoming requests, even if the previous requests failed.

Reviewer: see style ref. entry for "complete":
```suggestion
Until the recaching is complete, the expired JWKS is used to process the request, and a `503` error is returned if no request is found. This enables you to use a cached value in upcoming requests, even if the previous requests failed.
```
or
```suggestion
Until the recaching finishes, the expired JWKS is used to process the request, and a `503` error is returned if no request is found. This enables you to use a cached value in upcoming requests, even if the previous requests failed.
```

Reviewer: Can you clarify to what "This" refers in "This enables you...": is it the fact that the expired JWKS is used, or that the error is returned, or both?

Author: Done! Thanks!

|===
| Element | Description | Example
| JWT Origin | Specifies from where in the request the JWT will be extracted:
* HTTP Bearer Authentication Header
* Custom Expression

> If you set it to HTTP Bearer Authentication Header, the JWT is expected as Bearer. +

Reviewer: To what does "it" refer?
And if you are "setting" it using syntax, then it should be formatted as user-supplied syntax.

Author: Done! Thanks!

> If you set this field to *Custom Expression*, you must provide a DataWeave Expression returning the token.

Reviewer: ditto previous comment about format, `Custom Expression`.

Author: Done! Thanks!

| JWT Key Origin
| Specifies where to obtain the key for the Signature validation. +

> You can provide the key in the policy selecting the *Text* option or obtain it from a *JWKS*. +

Reviewer: ```suggestion
You can provide the key in the policy by selecting the *Text* option or by obtaining it from a JWKS. +
```

Reviewer: Question: if JWKS = ...Sets, plural, can we actually refer to "a" JWKS? Is it supposed to be Set, singular?

I see both when I search:

JSON Web Key Sets - Auth0https://auth0.com › Docs › Tokens › JSON Web Tokens
JSON Web Key Sets ... The JSON Web Key Set (JWKS) is a set of keys containing the public keys used to verify any JSON Web Token (JWT) issued by the ...
‎JSON Web Key Set Properties · ‎Locate JSON Web Key Sets · ‎Signing Keys

Author: Done! Thanks!

Ignore this field if you selected *none* as JWT Signing Method.
| This field input is the amount of time, in minutes, during which the policy considers the JWKS valid.

| JWKS Service connection timeout (milliseconds)

> | Sets the maximum time, in milliseconds, to wait for a response when authenticating the Access Token validation endpoint. +

Reviewer: Should Access Token be lowercase here?

Author: Done! Thanks!

> The default value is set to 10 seconds.

Reviewer: ```suggestion
The default value is 10 seconds.
```

Author: Done! Thanks!
